src/java.naming/share/classes/com/sun/naming/internal/ObjectFactoriesFilter.java

```diff
@@ -50,19 +50,58 @@ public final class ObjectFactoriesFilter {
     /**
      * Checks if serial filter configured with {@code "jdk.jndi.object.factoriesFilter"}
      * system property value allows instantiation of the specified objects factory class.
-     * If the filter result is not {@linkplain Status#REJECTED REJECTED}, the filter will
+     * If the filter result is {@linkplain Status#ALLOWED ALLOWED}, the filter will
      * allow the instantiation of objects factory class.
      *
-     * @param factoryClass objects factory class
+     * @param serialClass objects factory class
      * @return true - if the factory is allowed to be instantiated; false - otherwise
      */
-    public static boolean canInstantiateObjectsFactory(Class<?> factoryClass) {
-        return checkInput(() -> factoryClass);
+    public static boolean checkGlobalFilter(Class<?> serialClass) {
```

Comment:
I think the `serialClass` param should be renamed to `factoryClass` or something like that, since I think the `serialClass` reference comes from serialization/deserialization usage.

Reply:
The name comes from `ObjectInputFilter.FilterInfo` - it's been renamed from `factoryClass`to make it clear that the supplied lambda fills-in the non-default `ObjectInputFilter.FilterInfo.serialClass()` method.

Reply:
Thank you for that clarification, Aleksei. This sounds fine to me then.